students/Tianx/Lesson5/mailroom3.py

```diff
@@ -0,0 +1,189 @@
+# ------------------------------------------#
+# !/usr/bin/env python3
+# Title: mailroom3.py
+# Desc: Mailroom Part 3
+# Tian Xie, 2020-04-12, Created File
+# Tian Xie, 2020-04-26, Adding exception handler
+# ------------------------------------------#
+
+# A dictionary of donor paired with a list of donation amounts
+dict_of_donors = {'Jeff Bezos': [1.00, 50.00],
+                  'Warren Buffet': [100.00, 1000.00],
+                  'Bill Gates': [100.00, 500.00],
+                  'Tim Cook': [300.00],
+                  'Jack Ma': [2000.00]}
+# Main Prompt
+main_prompt = "\n".join(("=======  Main Menu ======= \n",
+                         "Welcome to the Mailroom. Please select an option:",
+                         "1: Send a Thank You to a single donor",
+                         "2: Create a Report",
+                         "3: Send letters to all donors",
+                         "4: Exit",
+                         ">>> "))
+
+
+def menu_selection(prompt, dispatch_dict):
+    """Displays a menu of choices to the user
+    Args:
+        prompt: main menu
+        dispatch_dict: choices for user to choose from
+    Returns:
+        None.
+    """
+    while True:
+        response = input(prompt)  # continuously collect user selection
+        while True:
+            try:
+                int(response)
+                break
+            except ValueError:
+                response = input('Error: Please enter a number. Please select again >')
+        if dispatch_dict[response]() == "Exit Menu":
+            break
```

Comment:
above needs some exception handling, what happens if you enter "9" as an option?